docs-src/design-decisions.md

```diff
@@ -0,0 +1,42 @@
+# Design Decisions
+
+Throughout the design of these APIs some significant design decisions have been
+made. This provides some background on the factors considered when making these
+decisions.
+
+## GatewayClass Scope
+
+### Problem
+There were a number of compelling reasons to make GatewayClass a
+namespace-scoped resource. 
+
+### Advantages of a Namespace-Scoped GatewayClass
+* Less potential for naming conflicts among different implementations.
+* Better modeling for implementations with multiple GatewayClass resources
+  (including some potential multi-tenant scenarios using Cluster API).
+* Easier to provide simple and secure defaults (only allow Gateways and Routes
+  in the same namespace by default).
+* GatewayClass parameters can live in the same namespace.
+* More RBAC flexibility (can allow users to create GatewayClasses in select
+  namespaces only or restrict visibility of GatewayClasses in some namespaces).
+
+### Advantages of a Cluster-Scoped GatewayClass
+* Consistent with StorageClass and IngressClass resources.
```

Comment:
Did these designs account for multi-tenancy where tenants may have access to their own infrastructure?

Reply:
I can't speak for StorageClass, but I don't believe this was a consideration for IngressClass.

Reply:
Same question here (ingress-nginx). Most of the users don't have enough permissions to create cluster-scoped resources and are limited to their own namespace.

Reply:
A "tenant" is undefined.  Before I will engage in any discussion of tenancy, you must tell me what that word means to you.